### PR TITLE
Add tribal query option

### PR DESCRIPTION
**Pull request by mbrousil**
Hey KW & B,

This is aimed at addressing https://github.com/USEPA/EPATADA/issues/345, querying tribal data.

The background here is that the EPA has a tribal land boundary map service, which EPATADA would like to use to provide access to tribal spatial data for WQP queries. See the EPA [guidance page](https://www.epa.gov/geospatial/guidance-using-tribal-boundaries-areas-and-names-resources) and the [Map Service page](https://geopub.epa.gov/arcgis/rest/services/EMEF/Tribal/MapServer).

This service has six layers of available data, which I found could be accessed with the [{arcgislayers}](https://github.com/R-ArcGIS/arcgislayers) package. The layers are: Alaska Native Allotments, Alaska Native Villages, American Indian Reservations, Off-reservation Trust Lands, Oklahoma Tribal Statistical Areas, and Virginia Federally Recognized Tribes.

What I've aimed to do here is: 
1. Allow the user to indicate which of the six layers to use
2. Filter the layer based on a user-provided tribe name or parcel id
3. Return an sf object of the corresponding spatial data inside the function
4. Proceed with the tribal sf object as though it was an sf object submitted via the `aoi_sf` function argument

This mostly has worked out, but there's a couple of potential snags:
+ It feels a bit clunky to have to have the user provide two arguments for this (not sure of an alternative though)
+ There mostly are **not** good columns with unique, externally meaningful identifiers in the sf objects returned from the Map Service. e.g., `NAME` values are repeated in a layer, `TRIBE_NAME` is unique but can be **very** long, `BIA_CODE` isn't reliably filled, `EPA_ID` isn't unique, `PARCEL_NO` isn't unique but this seems to be less of a concern compared with `NAME`
+ Alaska Native Villages is point data not shapefiles

But, I've implemented this and am hoping for feedback on whether the approach makes sense, whether there's any good alternatives you might be able to think of, and any other edits or suggestions you might have. Here's an example request for data from the American Indian Reservations layer:

```r
 TADA_DataRetrieval_test(
  tribal_area_type = "American Indian Reservations",
  tribe_name_parcel = "Spokane Tribe of the Spokane Reservation"
)
```

A couple of the layer appear not to correspond to any data in WQP: Alaska Native Villages & Virginia Federally Recognized Tribes. Off-reservation Trust Lands ***seems*** to work but might be returning bboxes too large to run successful queries.

I've also added a function, `TADA_TribalOptions()`, which returns either an sf object or data frame containing from the Map Service so that the user can review name/parcel options and details about capitalization, spelling, etc.

```r
TADA_TribalOptions("Virginia Federally Recognized Tribes")
```

Thanks and let me know if you have any questions about this!

### CODE REVIEW

File: src/TADA_TribalOptions.R
Comment (steeleb) [inline src/TADA_TribalOptions.R:1]: I think having a quick map option in here would be great, as simple as:

```
TADA_TribalOptions <- function(tribal_area_type, show_map = FALSE, return_sf = FALSE){
   ...
  # Show map if needed
  if(show_map){
    tmap::tmap_mode("view")
    names <- dplyr::if_else(tribal_area_type == "Alaska Native Allotments", "PARCEL_NO", "TRIBE_NAME")
    tribal_area_sf <- tribal_area_sf %>% sf::st_make_valid()
    if (grepl("POLYGON", unique(sf::st_geometry_type(tribal_area_sf)))) {
      map <- tmap::tm_shape(tribal_area_sf) + tmap::tm_polygons(id = names)
    } else {
      map <- tmap::tm_shape(tribal_area_sf) + tmap::tm_dots(id = names)
    }
    return(map)
  }
```
Obvi need to have handling for show_map and the dataframe/sf return.

File: src/TADA_TribalOptions.R [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                                  tribal_area == tribal_area_type)$url %>%
+    arcgislayers::arc_open() %>%
+    # Return sf
+    arcgislayers::arc_select()
+  
+  # Convert to df if needed, export
+  if(return_sf == FALSE){
+    return(
+      as.data.frame(tribal_area_sf) %>%
+        dplyr::select(-geometry)
```
Comment (steeleb) [inline src/TADA_TribalOptions.R]: An alternative:

```
      tribal_area_sf %>%
        sf::st_drop_geometry()
```

**Review by steeleb — COMMENTED** [review]
Nice!

High-level comment: throw a `sf::st_make_valid()` right before returning any {sf} object and/or before throw in the dataRetrieval pull. There are a few geometries from the Map Service that seem to be "invalid" in {sf} and that should fix them up to not cause downstream errors. 

For Virginia Federally Recognized Tribes, I would just provide a generic message of: "No land is associated with the selected tribe" per this info in the guidance doc:

> ### Virginia Federally Recognized Tribes
> Virginia Federally Recognized Tribes contains point data for federally recognized tribes geographically located in the state of Virginia, corresponding with the administrative boundaries of EPA Region 3. These tribes are federally recognized, but do not currently have land that qualifies as Indian country. Locations are based on mailing addresses for each tribe provided by the EPA Region 3 Tribal Program Coordinator to the US EPA Office of Mission Support on May 5, 2021.

For Alaska Native Villages, these are the centroid of an existing geometry. 184 of  227 can be found using the tigris::native_areas() sf:

```
native_areas <- tigris::native_areas()
EPA_points_with_tigris_area <- alaska_native_villages[native_areas,]
```
Maybe there is an AK boundary file somewhere that would have more of these? It doesn't look like the rest of the {tigris} options fit the bill either. 

For this:
>There mostly are not good columns with unique, externally meaningful identifiers in the sf objects returned from the Map Service. e.g., NAME values are repeated in a layer, TRIBE_NAME is unique but can be very long, BIA_CODE isn't reliably filled, EPA_ID isn't unique, PARCEL_NO isn't unique but this seems to be less of a concern compared with NAME

Oof. This is one where having an interactive/Shiny situation would probably be better, but alas, here we are. I think your gut to use either the PARCEL_NO or TRIBE_NAME is the right move here. Using some sort of text handler like {stringdist} might help here to find the 'closest' string if folks don't copy/paste from the `TADA_TribalOptions()` (but probably should include some error handling/messaging alongside it). 

I think adding a map option to the `TADA_TribalOptions()` would be advisable to make this even more user freindly. I provided a quick code snippet for this.

**Comment by kathryn-willi** [comment]
I think for those point feature layers, we could just provide a message saying "Polygon features for `tribal_area_type` are not available" and leave it at that for now...?

I also think that the requirement of the two tribal type/name arguments is annoying, but feel like there isn't any good solution for that. Perhaps we set the default to "American Indian Reservations" so it is less work for the majority of user queries?

**Pull request merged**